src/App.tsx

```diff
@@ -1,11 +1,60 @@
-import { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
+import { getMovie, NormalizeMovieData } from './api';
+import { MovieData } from './types/MovieData';
 
-export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState(false);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(
+      setAppliedQuery,
+      1000,
+    ),
+    [appliedQuery],
+  );
+
+  const handleSubmit = () => {
+    setIsLoading(true);
+    getMovie(appliedQuery)
+      .then((data) => {
+        if ((data as MovieData).imdbID) {
+          setMovie(NormalizeMovieData(data as MovieData));
+        }
+
+        setErrorMessage(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const addMovie = (addedMovie: Movie) => {
+    if (movies.some((element) => element.imdbId === addedMovie.imdbId)) {
+      setQuery('');
+      setMovie(null);
+
+      return;
+    }
```

Comment:
would be great to show some error in that case

Reply:
Done!